### DISCUSSION TITLE
custom ByteArrayFetcher not working as expected

### DISCUSSION THREAD

**Discussion by mirjalal** [coil-kt/coil#1200] [Q&A] [answered]
In v1.4.0 I have copied custom `ByteArrayFetcher` class [from here](https://github.com/coil-kt/coil/issues/171#issuecomment-554109785). As the lib updated & some classes have been removed, I tried to rewrite & use it. Currently it looks like this:
```
class ByteArrayFetcher(
    private val data: ByteArray,
    private val options: Options
) : Fetcher {
    override suspend fun fetch(): FetchResult =
        SourceResult(
            source = ImageSource(
                source = ByteArrayInputStream(data).source().buffer(),
                context = options.context,
            ),
            mimeType = null,
            dataSource = DataSource.MEMORY
        )

    class Factory : Fetcher.Factory<ByteArray> {
        override fun create(data: ByteArray, options: Options, imageLoader: ImageLoader): Fetcher =
            ByteArrayFetcher(data, options)
    }
}
```
I added it to `ImageLoader` components like:
```
ImageLoader.Builder(context)
    .components {
        add(SvgDecoder.Factory())
        add(ByteArrayFetcher.Factory())
    }
    .build()
```
But it didn't help. Then tried to add it on each `ImageRequest` like:
```
ImageRequest.Builder(LocalContext.current)
    .data(dataModel)
    .fetcherFactory(ByteArrayFetcher.Factory())
    .transformations(*transformations)
    .build()
```
Still it cannot show the image. 
p.s. in v1.4.0 there was no problem with the previous loader.

Did anybody faced & solved that (kind of) issue?

**Comment by colinrtwhite** [comment, marked as answer]
Hmm your `Fetcher` class does look correct. I'd check what error you're getting using `listener(onError = {})`.

Also Coil 2.0 adds support for `ByteBuffer`s so you could also wrap the array using `ByteBuffer.wrap(array)` then you don't need the custom `Fetcher`.

**Reply by mirjalal** [reply]
thanks. ```ByteBuffer.wrap(array)``` solved my problem.
